sanic/request.py

```diff
@@ -819,6 +820,15 @@ def url_for(self, view_name: str, **kwargs) -> str:
             view_name, _external=True, _scheme=scheme, _server=netloc, **kwargs
         )
 
+    @property
+    def scope(self) -> Optional[ASGIScope]:
+        """
+        :return: The ASGI scope of the request. If the app isn't an ASGI app, then returns None.
+        :rtype: Optional[ASGIScope]
+        """
+
+        return self.transport.scope if self.app.asgi else None
```

Comment:
I would rather this always return `ASGIScope` without `Optional`. It will be easier for annotations and not having to add something like the following whenever you want to use `request.scope.get`, etc.

```py
if request.scope:
    ...
```

Reply:
Instead, as mentioned in [my comment](https://github.com/sanic-org/sanic/issues/2417#issuecomment-1101342575) I would rather raise an exception if not in ASGI.